llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -2205,7 +2205,24 @@ void SelectionDAGLegalize::ExpandFPLibCall(SDNode* Node,
     Results.push_back(Tmp.first);
     Results.push_back(Tmp.second);
   } else {
-    SDValue Tmp = ExpandLibCall(LC, Node, false).first;
+    bool IsSignedArgument = false;
+    switch (LC) {
+    case RTLIB::LDEXP_F32:
+    case RTLIB::LDEXP_F64:
+    case RTLIB::LDEXP_F80:
+    case RTLIB::LDEXP_F128:
+    case RTLIB::LDEXP_PPCF128:
+      IsSignedArgument = true;
+      break;
+    case RTLIB::FREXP_F32:
+    case RTLIB::FREXP_F64:
+    case RTLIB::FREXP_F80:
+    case RTLIB::FREXP_F128:
+    case RTLIB::FREXP_PPCF128:
+      IsSignedArgument = true;
+      break;
```

Comment:
lgtm, except the frexp case looks untested

Reply:
There's a test for it in llvm/test/CodeGen/PowerPC/negintfplibcall.ll. Though now that I look at it more closely, why do we need to handle frexp here? It only has a floating-point arg, not an integer arg like ldexp.

Reply:
It has an int* out argument, but not sure how that would come in here

Reply:
The flag does apply to return values as well, but not to pointer out argument.

But in any case, FREXP goes through a different expansion method (ExpandFrexpLibCall), so handling it here doesn't really make sense to me.

Reply:
Fair enough.  In the interest of getting this merged, I removed the `frexp` changes -- `ldexp` is the main problem here and what is causing the Chromium problems.